### ISSUE TITLE
Allow specifying plugins as unicode (text) in python2

### ISSUE THREAD

**Issue by reutsharabani** [pytest-dev/pytest#2344]
Currently `config.py:398` enforces plugin names to be strings when specified in `conftest.py`. I'd like for it to support unicode file names (for example when using `from __future__ import unicode literals`).

I couldn't find a ticket for it. Here is an example:

Create `conftest.py` file:

    reut@reut-Latitude-E5450:~/projects/example$ cat conftest.py 
    pytest_plugins = u"123"

Run pytest:

    reut@reut-Latitude-E5450:~/projects/example$ pytest 2>&1 | tail
        self.consider_conftest(mod)
      File "/home/reut/miniconda2/lib/python2.7/site-packages/_pytest/config.py", line 374, in consider_conftest
        self.consider_module(conftestmodule)
      File "/home/reut/miniconda2/lib/python2.7/site-packages/_pytest/config.py", line 384, in consider_module
        self._import_plugin_specs(plugins)
      File "/home/reut/miniconda2/lib/python2.7/site-packages/_pytest/config.py", line 391, in _import_plugin_specs
        self.import_plugin(import_spec)
      File "/home/reut/miniconda2/lib/python2.7/site-packages/_pytest/config.py", line 398, in import_plugin
        assert isinstance(modname, str)
AssertionError

I suggest adding `py.builtin.text` as a second allowed type, but you may have a better solution. I can create a PR if you have a suggestion I can implement.

I may also need help adding a test for it (I suspect it is easy, but browsing `testing/test.config.py` didn't reveal how).

Thanks.

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus any idea if this might cause severe issues?

**Comment by nicoddemus** [comment]
@RonnyPfannschmidt I can't think of any beside someone passing a unicode string which is not a module and contains non-ascii characters, which would blow up quite spectacularly. 😉 

But we are already at risk today so I don't think this will hurt at all, please @reutsharabani feel free to submit a PR to the `features` branch.

FWIW at work we adopt the policy of importing `unicode_literals` in all modules, and while a little annoying we use this workaround which works in Python 2 and 3:

```python
from __future__ import unicode_literals

pytest_plugins = str('ben10.fixtures')
```

**Comment by nicoddemus** [comment]
Resolved in #2347
